src/lib/eina/eina_file_win32.c

```diff
@@ -917,24 +917,35 @@ eina_file_map_new(Eina_File *file, Eina_File_Populate rule,
    if (!map)
      {
         HANDLE fm;
+        SYSTEM_INFO sys_info;
+
+        GetSystemInfo(&sys_info);
+
         map = malloc(sizeof (Eina_File_Map));
         if (!map)
           {
              eina_lock_release(&file->lock);
              return NULL;
           }
 
+        const DWORD align = ~(sys_info.dwAllocationGranularity - 1);
+        const DWORD aligned_offset = offset & align;
+        const DWORD offset_diff = offset - aligned_offset;
+        const uint64_t total_length = offset + length;
+        const DWORD length_low = total_length & (uint64_t) (DWORD)-1;
+        const DWORD length_high = total_length >> (sizeof(DWORD) * 8);
+
```

Comment:
Could you explain a little what is happening here? I couldn't get it.

Reply:
The typical alignment granularity for Windows is 64kb. That means the `MapViewOfFile` offset must be a multiple of 64kb. To return the proper pointer, we must find the offset that is aligned by 64kb that is equal or less than the passed offset. `align` is the mask to find the aligned offset. `aligned_offset` is the final offset value. `offset_set` is the distance from the `aligned_offset` to the passed `offset`.